### DISCUSSION TITLE
After the upgrade from 3.8.1 to 4.4.1 - Error Message - ER7Serializer error ERROR MESSAGE

### DISCUSSION THREAD

**Discussion by Jaysarun** [nextgenhealthcare/connect#6092] [Q&A]
Hi Guys,
I am currently facing this below issue, Whenever I try to send a message to the channel.

ER7Serializer error
ERROR MESSAGE: Error converting XML to ER7
org.xml.sax.SAXParseException; Premature end of file.

Attaching the screenshot below,

![image](https://github.com/nextgenhealthcare/connect/assets/155644178/0139bc9f-d595-4242-9d0f-c5b83301a783)

Any suggestion on this one?

**Comment by jonbartels** [comment]
That error is typically a problem with:
* Sending the wrong datatype to the channel
* Missing start/end characters or a typo in the message
* The wrong datatype selection in the channel

**Reply by Jaysarun** [reply]
Noted @jonbartels, I will check those.

**Reply by Jaysarun** [reply]
Hi @jonbartels,

I have checked the above points. It all looks good. The Same message was able to process in the old version (3.8.1) but it only causing "ER7Serializer error" in the new version (4.4.1). Any other things I need to look into it?

**Reply by jonbartels** [reply]
Post a sample message. (Attach as text file or please use code formatting!)
Screenshot your data type properties.
Post any code that your channel uses to serialize or deserialize messages.

With an issue that only appears for an upgrade, check the release notes and see if a parsing default changed.

**Comment by Jaysarun** [comment]
Hi @jonbartels,

I have removed all of my codes and just added a filter step and try to catch a values coming in the MSH4 Segment but still I am getting the same error. Also I have validated the points you have mentioned above and all looks fine. Could you please share your thoughts?

**Here is the sample code,**
```JS
var zHCOID1 = 50003; // Declared the facility ID here

// PUT FACILITY HCOID HERE END
if ((msg['MSH']['MSH.4']['MSH.4.1'].toString() == zHCOID1)) //capturing the value here in this segment
{
	return true;	
} else
{
	return false;
}
```


**sample message:**
```
MSH|^~\&|SENDING_APPLICATION|50003|RECEIVING_APPLICATION|RECEIVING_FACILITY|20110613083617||ADT^A04|934576120110613083617|P|2.3||||
EVN|A04|20110613083617|||
PID|1||135769||MOUSE^MICKEY^||19281118|M|||123 Main St.^^Lake Buena Vista^FL^32830||(407)939-1289^^^theMainMouse@disney.com|||||1719|99999999||||||||||||||||||||
PV1|1|O|||||7^Disney^Walt^^MD^^^^|||||||||||||||||||||||||||||||||||||||||||||
```

**Attaching the error message**
![image](https://github.com/nextgenhealthcare/connect/assets/155644178/59004e01-be72-483a-ac3b-10c0319f7ece)

**Reply by pacmano1** [reply]
@Jaysarun please use code blocks and format your code, read https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks.

Fix your post please.

**Reply by pacmano1** [reply]
If fixed it for you actually, next time please use code blocks.

**Comment by Jaysarun** [comment]
Hi @pacmano1,
Apologies, I have tried the formatting now. could you please check and let me know if anything is wrong.

var zHCOID1 = 50003; // Declared the facility ID here

// PUT FACILITY HCOID HERE END
if ((msg['MSH']['MSH.4']['MSH.4.1'].toString() == zHCOID1)) { //capturing the value here in this segment
    return true;
} else {
    return false;
}

**Reply by pacmano1** [reply]
@Jaysarun Please read the link I posted: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

Please look at your original post and edit it and see how I formatted it for you and used code blocks.

For example:

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/2bc22db8-8470-44c0-aa05-e134b279d4ea)

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/1d41743f-8593-4af0-9a5f-e407c3d14233)

**Reply by pacmano1** [reply]
Please read what people write here:

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/ca28ff9a-0e54-40c4-80d9-99d4a4b21305)

**Reply by Jaysarun** [reply]
I am so sorry about that, I will share the code as per the standards. Thanks @pacmano1

**Reply by pacmano1** [reply]
No problem at all.   On your original question here, I suspect the message may have something like a BOM (byte order mark) or some other "special" character.  You can test that by setting up a channel that does nothing at all.  i.e. set up a TCP listener (or file reader, however you are getting the message) with no filters and no transformers, leave the destination a channel reader.  Send a message to it.  If it fails it usually means you have some special character in the message itself.   You need to view the raw messaage in an editor that can display special characters.   

It can be something as simple as messing with the below, but if what you posted above is a full mesage, I suspect is is not related to the screenshot.

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/5cb46a1d-ff95-4406-b8f5-48acf108abcc)

**Reply by pacmano1** [reply]
Also check your outbound message template (in your original channel).   

It might actually be best to upload a copy of your channel here after:

1) making sure there is no PII/PHI (people forget to remove message templates sometimes)
2) no other confidential stuff.

**Reply by Jaysarun** [reply]
Thanks for you support, Appreciate it @pacmano1 ! I have doubted the same thing as you have mentioned above then I created a channel without any filters and transformations. I was able to process the same ADT messages with out any issues. whenever I am trying with the Javascripts code for Filters and transformations that is where the parser error is coming. In the old version 3.8.1 everything is running based on our JS code, I checked the release note and not much of a change in the new version. Being stuck at this issue for a while now.

**Reply by pacmano1** [reply]
post an export of your channel

with code templates included, of course

**Reply by Jaysarun** [reply]
Sure @pacmano1 , I am attaching the channel now. To read the DB connection we are using a Jar file in the Cust-lib folder. I can't share it with you since it got the DB details. Please let me know if you need anything else from my end. Thanks for your Support and Guidance.

[Test training.zip](https://github.com/nextgenhealthcare/connect/files/14394479/Test.training.zip)

**Reply by pacmano1** [reply]
I don't see anything off hand that would break the message.   But I obviously can't fully test your channel.

Try this:  Move the jar outside of the custom-lib folder and instead add it as a resource for your channel under settings..resources.  Check the box that says "Load Parent First".   You may need to restart Mirth.  Curious if something in your custom jar is now broken with a newer version of Mirth.   

Actually move all jars out of custom-lib.

**Reply by Jaysarun** [reply]
Hi @pacmano1 Noted. I will try it and let you know. Thank you!
